crud/select/plan.lua

```diff
@@ -165,7 +165,9 @@ function select_plan.new(space, conditions, opts)
     -- get sharding key value
     local sharding_key
     if scan_value ~= nil and (scan_iter == box.index.EQ or scan_iter == box.index.REQ) then
-        sharding_key = extract_sharding_key_from_scan_key(scan_value, scan_index, sharding_index)
+        if #scan_value == #scan_index.parts then
```

Comment:
Why do you add new nesting level. Seems you can specify this condition at previous line.